Week1/credit/credit.c

```diff
@@ -0,0 +1,85 @@
+#include <cs50.h>
+#include <stdio.h>
+
+int main(void)
+{
+    // input creditcard number
+    long n = get_long("Creditcardnumber: ");
+
+    // Count lenght of Creditcardnumber
+    int i = 0;
+    long cc = n;
+    while (cc > 0)
+    {
+        cc = cc / 10;
+        i++;
+    }
+
+    // Check the lenght
+    if (i != 13 && i != 15 && i != 16)
+    {
+        printf("INVALID\n");
+        return 0;
+    }
+
+    // Calculate checksum
+    int sum1 = 0;
+    int sum2 = 0;
+    long x = n;
+    int total = 0;
+    int mod1;
+    int mod2;
+    int d1;
+    int d2;
+    do
+    {
+        // Remove last digit and add to sum1
+        mod1 = x % 10;
+        x = x / 10;
+        sum1 = sum1 + mod1;
+        // Remove second last digit
+        mod2 = x % 10;
+        x = x / 10;
+        // Double second last digit and add digits to sum2
+        mod2 = mod2 * 2;
+        d1 = mod2 % 10;
+        d2 = mod2 / 10;
+        sum2 = sum2 + d1 + d2;
+    } while (x > 0);
+    total = sum1 + sum2;
+
+    // checking Invalid
+    if (total % 10 != 0)
+    {
+        printf("INVALID\n");
+        return 0;
+    }
+
+    // Get starting digits
```

Comment:
I think this kind of comment is ok because it tries to structure the programm here.
But to practice for later projects please remove those comments as well :)